internal/pkg/agent/application/upgrade/crash_checker_test.go

```diff
@@ -123,7 +125,35 @@ func TestChecker(t *testing.T) {
 		}
 
 		canc()
-		require.Error(t, err)
+		require.ErrorContains(t, err, "service restarted '3' times within '0.1' seconds")
+	})
+
+	t.Run("fails when pid remains 0", func(t *testing.T) {
+		const startingPID = 0
+		pider := &testPider{pid: startingPID}
+		ch, errChan := testableChecker(t, pider)
+		ctx, canc := context.WithCancel(context.Background())
+
+		var wg sync.WaitGroup
+		wg.Add(1)
+		go func() {
+			wg.Done()
+			ch.Run(ctx)
+		}()
+
+		wg.Wait()
+		for i := 0; i < 3; i++ {
+			<-time.After(testCheckPeriod * 3)
+			pider.Change(startingPID) // don't change PID
+		}
+		var err error
+		select {
+		case err = <-errChan:
+		default:
+		}
+
+		canc()
+		require.ErrorContains(t, err, "service remained crashed (PID = 0) within '0.1' seconds")
```

Comment:
[NIT]
I tend to reserve `require` to the cases when a failure would lead to unexpected behaviour further down the test. Otherwise I understand `assert` is more appropriated also because, on tests with more assertions, all the assertions run and the error shows everything differing from the expected.